src/model_theory/syntax.lean

```diff
@@ -66,13 +66,60 @@ namespace term
 
 open list
 
+/-- Indicates whether a variable is used in a given term. -/
+@[simp] def uses_var : L.term α → α → Prop
+| (var i) a := i = a
+| (func f ts) a := ∃ i, (ts i).uses_var a
+
+lemma var_uses_var (a : α) : (var a : L.term α).uses_var a :=
+by rw uses_var
+
+lemma func_uses_var {n : ℕ} {f : L.functions n} {a : α} {ts : fin n → L.term α} {i : fin n}
```

Comment:
```suggestion
lemma uses_var.func {n : ℕ} {f : L.functions n} {a : α} {ts : fin n → L.term α} {i : fin n}
```
also consider making `i` explicit. `i` might be hard to infer from the type of `h` (if `ts` is not a variable)